### PR TITLE
[compaction]refact compaction work operator for supporting multi unaware bucket tables.

### PR DESCRIPTION
**Pull request by wg1026688210**
related #2670
### Purpose

Linked issue: close #xxx

### Tests
AppendOnlyMultiTableCompactionWorkerOperatorTest.java for test AppendOnlyMultiTableCompactionWorkerOperator

### CODE REVIEW

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/compact/UnwareBucketCompactionHelper.java [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Future;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+
+/**
+ * The help class for the Compaction of unware bucket append table to execute {@link
+ * AppendOnlyCompactionTask}.
+ */
+public class UnwareBucketCompactionHelper {
```
Comment (leaves12138) [inline paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/compact/UnwareBucketCompactionHelper.java:45]: `UnwareBucketCompactor` ? This class actually does the compaction job, so it is the real compactor.

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/compact/UnwareBucketCompactionHelper.java [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+            String commitUser,
+            Supplier<ExecutorService> lazyCompactExecutor) {
+        this.table = table;
+        this.commitUser = commitUser;
+        this.write = (AppendOnlyFileStoreWrite) table.store().newWrite(commitUser);
+        this.result = new LinkedList<>();
+        this.compactExecutorsupplier = lazyCompactExecutor;
+    }
+
+    public void processElement(StreamRecord<AppendOnlyCompactionTask> element) throws Exception {
```
Comment (leaves12138) [inline paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/compact/UnwareBucketCompactionHelper.java:66]: `public void submit(AppendOnlyCompactionTask task) throws Exception`?  Compactor does not need to access `StreamRecord`, just pass `AppendOnlyCompactionTask` to it is okay.

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/compact/UnwareBucketCompactionHelper.java (hunk 1 above)
Comment (wg1026688210) [reply paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/compact/UnwareBucketCompactionHelper.java]: done

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/compact/UnwareBucketCompactionHelper.java (hunk 2 above)
Comment (wg1026688210) [reply paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/compact/UnwareBucketCompactionHelper.java]: done

**Comment by wg1026688210** [comment]
finished at #2858

**Pull request closed without merging**